### PR TITLE
Update operator-sdk and Go version

### LINKED ISSUE #176: Update Operator SDK and Go version
**Is your feature request related to a problem? Please describe.**
Current version of the opendatahub-operator has outdated packages and sdk version

**Describe the solution you'd like**
- Update go version to v1.17.2
- Update operator-sdk version to latest
-  Update kubernetes version to v1.22
- Update any deprecated methods

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by VaishnaviHire**
Fixes #176 

## Description
-  Update Operator to run with latest SDK and Go version 1.17
-  Remove support for Kfdef v1alpha1, v1beta1

## How Has This Been Tested?
1. Openshift cluster 4.11
2.  Deployment instructions:
     -  [Deploy using OLM](https://github.com/opendatahub-io/opendatahub-operator/pull/184/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R52)
     -  [Deploy locally](https://github.com/opendatahub-io/opendatahub-operator/pull/184/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R40)

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by shalberd** [comment]
so neat, thank you for making that change, @VaishnaviHire . Now, with config/manager/manager.yaml present and OCP 4.10, it becomes much easier to make an oprator bundle supported a set of relatedImages, and without having to manually find out the sha256 digests. See https://docs.openshift.com/container-platform/4.10/operators/operator_sdk/osdk-generating-csvs.html#olm-enabling-operator-for-restricted-network_osdk-generating-csvs @VaishnaviHire

**Comment by VaishnaviHire** [comment]
Note: This PR is ready for reviews

File: Dockerfile [hunk 1]
```diff
@@ -1,78 +1,27 @@
...
-RUN mkdir kubeflow
-RUN echo REPO=${REPO} branch=${BRANCH}
-RUN git clone ${REPO} --depth=1 --branch ${BRANCH} --single-branch kubeflow
-WORKDIR ${GOPATH}/src/github.com/kubeflow/kfctl
-
-# Download dependencies first to optimize Docker caching.
-COPY go.mod .
-COPY go.sum .
+# Build the manager binary
+FROM golang:1.18 as builder
```
Comment (anishasthana) [inline Dockerfile:2]: Can we use one of the one of the ubi8/golang tool images?
I'm seeing a bunch at

```
❯ skopeo list-tags docker://registry.access.redhat.com/ubi8/go-toolset | grep 1.18
        "1.18",
        "1.18.4",
        "1.14.12-17.1618436992-source",
        "1.18.4-8.1669838000-source",
        "1.14.12-17.1618436992",
        "1.18.4-8-source",
        "1.18.4-8.1669838000",
        "1.18.4-8",
```

**Comment by VaishnaviHire** [comment]
Note: Updated the base branch to `feature/update-sdk`

File: Dockerfile (hunk 1 above)
Comment (anishasthana) [reply Dockerfile:2]: Could you actually use registry.redhat.io/ubi8/go-toolset instead?

Comment (VaishnaviHire) [reply Dockerfile]: Update the Dockerfile with `registry.redhat.io/ubi8/go-toolset` image with Go version as a docker `ARG`

**Comment by VaishnaviHire** [comment]
Updated the PR with instructions to deploy Operator using OLM

File: Dockerfile [hunk 2]
```diff
@@ -1,78 +1,28 @@
...
+COPY controllers/ controllers/
+COPY pkg/ pkg/
 
-#**********************************************************************
-#
-# kfctl
-#
-FROM barebones_base as kfctl
+# Build
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
```
Comment (LaVLaS) [inline Dockerfile:20]: With the change to ubi8/go-toolset, this line is throwing an error
```
[1/2] STEP 10/10: RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go
controllers/kfdef.apps.kubeflow.org/kfdef_controller.go:61:2: open /opt/odh-operator/pkg/kfapp/coordinator: permission denied
controllers/kfdef.apps.kubeflow.org/kfdef_controller.go:62:2: open /opt/odh-operator/pkg/kfconfig/loaders: permission denied
controllers/kfdef.apps.kubeflow.org/kfdef_controller.go:63:2: open /opt/odh-operator/pkg/utils: permission denied
Error: building at STEP "RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o manager main.go": while running runtime: exit status 1
```

File: Makefile [hunk 3]
```diff
@@ -1,312 +1,239 @@
...
+build: generate fmt vet ## Build manager binary.
+	go build -o bin/manager main.go
+
+.PHONY: run
+run: manifests generate fmt vet ## Run a controller from your host.
+	go run ./main.go
+
+.PHONY: docker-build
+docker-build: manifests generate fmt vet ## Build docker image with the manager.
+	docker build -t ${IMG} .
```
Comment (LaVLaS) [inline Makefile:121]: I can do this in a follow-up PR but we should move use an `IMG_BUILDER` variable to allow someone to choose `podman` or `docker`

Comment (anishasthana) [reply Makefile:121]: We should also default to podman.

File: Dockerfile (hunk 2 above)
Comment (LaVLaS) [reply Dockerfile:20]: This is is due to the builder image running as the `default` user but the 'WORKDIR` `/workspace` has `755` permissions preventing the `go` command from creating files.

File: Dockerfile [hunk 4]
```diff
@@ -1,78 +1,28 @@
...
-WORKDIR ${GOPATH}/src/github.com/kubeflow/kfctl
-
-# Download dependencies first to optimize Docker caching.
-COPY go.mod .
-COPY go.sum .
+# Build the manager binary
+ARG GOLANG_VERSION=1.18.4
+FROM registry.access.redhat.com/ubi8/go-toolset:$GOLANG_VERSION as builder
+
+WORKDIR /workspace
```
Comment (LaVLaS) [inline Dockerfile:5]: Fix for `permission denied` in the `go build` command
```suggestion
WORKDIR /workspace

USER root
```

File: README.md [hunk 5]
```diff
@@ -1,5 +1,81 @@
...
+
+The latest version of operator can be installed from the community-operators catalog on OperatorHub. It can also be build 
+and installed from source manually, see the Developer guide for further instructions.
+
+### Developer Guide
+
+**Pre-requisites**
+
+- Go version **go1.18.4** 
+- operator-sdk version can be updated to **v1.26.0**
```
Comment (LaVLaS) [inline README.md:21]: Testing out this operator version the kubebuilder version in `v1.26.0` requires go 1.19+.

We should just set this to operator-sdk 1.24.1

File: Makefile (hunk 3 above)
Comment (LaVLaS) [reply Makefile]: @VaishnaviHire I can make this change in a follow-up PR if you want

**Review by LaVLaS — APPROVED** [review]
/lgtm

Build and deploy worked perfectly for me without any issues. I created a new kfdef and the operator deployed the pods without any noticeable issues

**Pull request merged**